README.md

```diff
@@ -4,45 +4,145 @@
 
 # QBoost
 
-The D-Wave quantum computer has been widely studied as a discrete optimization
-engine that accepts any problem formulated as quadratic unconstrained binary
-optimization (QUBO). In 2008, Google and D-Wave published a paper, [Training a
-Binary Classifier with the Quantum Adiabatic
-Algorithm](https://arxiv.org/pdf/0811.0416.pdf), which describes how the QBoost
-ensemble method makes binary classification amenable to quantum computing: the
-problem is formulated as a thresholded linear superposition of a set of weak
-classifiers and the D-Wave quantum computer is used to optimize the weights in a
-learning process that strives to minimize the training error and number of weak
-classifiers
-
-This code demonstrates the use of the D-Wave system to solve a binary
-classification problem using the QBoost algorithm.
+In machine learning, boosting methods are used to combine a set of simple,
+"weak" predictors in such a way as to produce a more powerful, "strong"
+predictor.  QBoost is an approach for boosting that involves selecting weak
+classifiers based on minimization of quadratic loss (squared difference between
+the observed and predicted label), so that the resulting optimization problem
+can be formulated as a binary quadratic model and solved on the D-Wave system.
+This approach has been explored previously in the literature, for example
+Refs. [1-4].
 
-## Disclaimer
-
-This demo and its code are intended for demonstrative purposes only and are not
-designed for performance.
+This example illustrates the application of a QBoost formulation towards two
+illustrative data sets.
 
 ## Usage
 
-To run a demonstration using one of two provided sample data sets, run:
+The demonstration includes two sample data sets, which can be run with the
+following commands:
 
 ```bash
-python demo.py  wisc
+python demo.py blobs
 ```
 
 Or:
 
 ```bash
-python demo.py mnist
+python demo.py digits
+```
+
+### Blobs data set
+
+This is a simple synthetically generated data set for illustrative purposes.  It
+is based on the scikit-learn function `sklearn.datasets.make_blobs`.  It
+generates a data set for classification with an arbitrary number of features,
+samples, and informative features.  The features are randomly generated from
+Gaussian distributions.  For each informative feature, the mean value differs
+between the two target classes, while the non-informative features are sampled
+from the same probability distribution in each class.
+
+Each run of the demo will randomly generate the data set and split it into
+training, validation, and test sets.  The validation set is used to determine
+the best value of the regularization parameter using a simple parameter sweep.
+The results of the demo will print the indices of the actual informative
+features, the features that are selected by QBoost, and the accuracy of the
+resulting ensemble classifier on the test set.
```

Comment:
The score is not the best achievable, it is the proportion of test set labels that a classifier predicts correctly.

Your suggestion in 2 is interesting.  I hadn't thought about doing this, but it is valid that one might be interested in that type of comparison.  However, for the purposes of this demonstration code, IMO this is not what we are *most* interested in.  My aim is to focus more on the intuitive aspects, especially with the "blobs" data set.  It is intuitive because we have a known set of informative features, and all other features provide no information and should in principle not be selected.  I would rather highlight the features that are selected vs the features that we know should be selected, as opposed to an accuracy comparison.  IMO interpreting the differences that we would get from the accuracy comparison would be less intuitive, and it would be subject to other details of the problem formulation, such as the means and variances of the features.

(There is also a second more subtle problem, which is that without a lot of care in how we set up such a comparison, we could very likely get accuracy values that differ between the ensemble of informative features and qboost, even when qboost selects exactly the informative features.  This could happen because of how the offset is fit, and training/validation splits, and it might depend on the value of the new `--cross-validation` option.  Obviously we would want to avoid this because it would be extremely confusing for the user to see such a comparison between two ensembles that are conceptually the same but produce different accuracy results.)